daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionCoder.scala

```diff
@@ -217,6 +217,7 @@ object TransactionCoder {
             .addAllChildren(ne.children.map(encodeNid.asString).toList.asJava)
             .addAllSignatories(ne.signatories.toSet[String].asJava)
             .addAllStakeholders(ne.stakeholders.toSet[String].asJava)
+            .addAllObservers(ne.observers.toSet[String].asJava)
```

Comment:
We should check the version here ! 

You have to add in `TransactionVersions`: 
```scala
private[transaction] val minChoiceObserver = TransactionVersion("11")
```

And somewhere around here: 
```scala
if (transactionVersion precedes minChoiceObserver && ne.observers.nonEmpty) 
  throw  EncodeError(
                s"As of version $minNoControllers, the observer of an exercise node _must_ be empty",
  )
```